engine/classes/ElggSession.php

```diff
@@ -384,4 +388,15 @@ public function offsetExists($offset) {
 
 		return false;
 	}
+
+	/**
+	 * Get an isolated ElggSession that does not persist between requests
+	 *
+	 * @return ElggSession
```

Comment:
Will do. To me `self` implies a dynamic method returning the instance it's bound to, but that's my imagination, not from phpdoc.org